src/dd_header.F90

```diff
@@ -0,0 +1,109 @@
+module dd_header
+
+  use dict_header,      only: DictIntInt
+  use mesh_header,      only: RegularMesh
+  use particle_header,  only: Particle, ParticleBuffer
+
+  implicit none
+  private
+  public :: deallocate_dd
+
+  type, public :: dd_type
```

Comment:
Use CamelCase for derived type names and no underscores. See the [style guide](http://openmc.readthedocs.org/en/latest/devguide/styleguide.html#derived-types-and-classes)